gcc/rust/ast/rust-ast-builder-base.h

```diff
@@ -0,0 +1,234 @@
+// Copyright (C) 2020-2024 Free Software Foundation, Inc.
+
+// This file is part of GCC.
+
+// GCC is free software; you can redistribute it and/or modify it under
+// the terms of the GNU General Public License as published by the Free
+// Software Foundation; either version 3, or (at your option) any later
+// version.
+
+// GCC is distributed in the hope that it will be useful, but WITHOUT ANY
+// WARRANTY; without even the implied warranty of MERCHANTABILITY or
+// FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+// for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with GCC; see the file COPYING3.  If not see
+// <http://www.gnu.org/licenses/>.
+
+#include "rust-ast-visitor.h"
+
+namespace Rust {
+namespace AST {
+
+// base class to allow derivatives to overload as needed
+class ASTBuilderBase : public AST::ASTVisitor
```

Comment:
Unless you do need empty visit functions, it would be better to inherit from `AST::DefaultASTVisitor`. Multiple classes are already using this and it is less prone to errors:
- Nodes are almost always reachable
- You can use `AST::DefaultVisitor::visit(whatever_you_were_visiting)` from within your overload once you've implemented your behavior to automatically visit child nodes.

Reply:
sounds good

Reply:
just removed that visitor to use the default one